File: 2015/puzzle1.py
# read puzzle1.dat as a string.
def read_puzzle():
    with open('puzzle1.dat', 'r') as f:
        return f.read().strip()

puzzle = read_puzzle()

print ('Part 1:', puzzle.count('(') - puzzle.count(')'))

floor = 0
for x in range(len(puzzle)):
    if puzzle[x] == '(':
        floor += 1
    elif puzzle[x] == ')':
        floor -= 1
    if floor == -1:
        print('Part 2:', x + 1)
        break

File: 2015/puzzle1.clj
(def data (slurp "puzzle1.dat"))
(println "There are " (count data) " characters in the file.")
(println "The first character is " (first data))
(println "The first ten characters are " (take 10 data))
(def freq (frequencies data))
(println "Part 1:" (- (freq \() (freq \))))
(comment "map ( to 1 and ) to -1")
(def freq2 (map (fn [x] (if (eq x ?\() 1 -1) x)) data))
(println "First ten mapped are " (take 10 freq2))
